torch/testing/_internal/distributed/distributed_test.py

```diff
@@ -789,6 +789,36 @@ def test_backend_group(self):
         def test_backend_full_group(self):
             self._test_group_override_backend(self._init_full_group_test)
 
+        @skip_if_lt_x_gpu(int(os.environ["WORLD_SIZE"]))
+        def test_new_default_subgroups_nccl(self):
+            world_size = dist.get_world_size()
+            cur_subgroup, subgroups = dist.new_subgroups()
+            # By default, since this test runs on a single machine, only one subgroup will be generated,
+            # which should be equal to the main process group that contains all the ranks.
+            self.assertEqual(cur_subgroup.size(), world_size)
+            self.assertEqual(len(subgroups), 1)
+            self.assertEqual(cur_subgroup, subgroups[0])
+            self.assertFalse(dist._rank_not_in_group(cur_subgroup))
+
+        @skip_if_lt_x_gpu(4)
+        def test_new_2subgroups_nccl(self):
```

Comment:
Can we also add a tests for:
1) malformed input, i.e. bad `ranks_per_subgroup_list`
2) validate logic when there is overlapping (i.e. rank occurs more than once in the sublists)
3) What happens when we pass in a rank into `rank_per_subgroup_list` that is not part of the passed in `group`?